Automattic/jetpack
# Add Resumable Upload Support to the VideoPress Block

Large video uploads can easily error out, requiring a user to completely start their upload over again. This PR adds a new component to the VideoPress block that uploads a video to wp.com via the [Tus](https://tus.io/) protocol. This enables automatic upload retries, and the ability to resume an upload even if a user closes their browser window.

<img width="629" alt="Screen Shot 2022-02-04 at 11 05 51 AM" src="https://user-images.githubusercontent.com/789137/152612848-0e0db910-d375-4928-ad59-3814eac06691.png">

#### Changes proposed in this Pull Request:
* Adds a new `resumable-upload` component that manages a resumable upload to a new API endpoint on wp.com.
* On wp.com only, we enable a window variable to enable the feature. Once we've tested it we can easily enable uploads for Jetpack in a future PR.

#### Jetpack product discussion
* pxWta-16I-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
**For Jetpack, we just verify existing functionality still works:**
* From the Jetpack dashboard, turn OFF VideoPress
* Open the block editor and add some videos to a post using the `Upload` button (or by drag and drop into the editor). The video should be uploaded to the sites media library and use the core video block.
* Enable VideoPress in the Jetpack dashboard
* Return to the block editor, and upload a new video. It should use the current VideoPress uploader (`Uploading...` displayed with a spinner).
* Verify `Media library` and `Insert from url` flows still work as expected.

**For wp.com:**
* See D74124-code for instructions on testing the block on a wp.com sandbox with the resumable uploader enabled.
